demo/admin/src/products/ManufacturerForm.tsx

```diff
@@ -0,0 +1,376 @@
+import { useApolloClient, useQuery } from "@apollo/client";
+import {
+    Field,
+    FieldSet,
+    FinalForm,
+    FinalFormInput,
+    FinalFormSaveSplitButton,
+    FinalFormSubmitEvent,
+    FormSection,
+    Loading,
+    MainContent,
+    TextField,
+    Toolbar,
+    ToolbarActions,
+    ToolbarFillSpace,
+    ToolbarItem,
+    ToolbarTitleItem,
+    useFormApiRef,
+    useStackApi,
+    useStackSwitchApi,
+} from "@comet/admin";
+import { ArrowLeft } from "@comet/admin-icons";
+import { EditPageLayout, queryUpdatedAt, resolveHasSaveConflict, useFormSaveConflict } from "@comet/cms-admin";
+import { IconButton } from "@mui/material";
+import { FormApi } from "final-form";
+import { filter } from "graphql-anywhere";
+import isEqual from "lodash.isequal";
+import React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { createManufacturerMutation, manufacturerFormFragment, manufacturerQuery, updateManufacturerMutation } from "./ManufacturerForm.gql";
+import {
+    GQLCreateManufacturerMutation,
+    GQLCreateManufacturerMutationVariables,
+    GQLManufacturerFormDetailsFragment,
+    GQLManufacturerQuery,
+    GQLManufacturerQueryVariables,
+    GQLUpdateManufacturerMutation,
+    GQLUpdateManufacturerMutationVariables,
+} from "./ManufacturerForm.gql.generated";
+
+type FormValues = Omit<GQLManufacturerFormDetailsFragment, "address" | "addressAsEmbeddable"> & {
+    address: {
+        street: string;
+        streetNumber: string | null;
+        zip: string;
+        country: string;
+        alternativeAddress: { street: string; streetNumber: string | null; zip: string; country: string };
+    };
+    addressAsEmbeddable: {
+        street: string;
```

Comment:
you should only specify the non-standard fields that needs changing.

And we want to try MergeDeep from type-fest for that

Reply:
I tried it but does not seem to work as expected, at least for nested

```
type FormValues = MergeDeep<
    GQLManufacturerFormDetailsFragment,
    {
        address: {
            streetNumber: string | null;
            zip: string;
            alternativeAddress: { streetNumber: string | null; zip: string };
        };
        addressAsEmbeddable: {
            streetNumber: string | null;
            zip: string;
            alternativeAddress: { streetNumber: string | null; zip: string };
        };
    }
>;
```
does result in this
<img width="835" alt="image" src="https://github.com/vivid-planet/comet/assets/1324250/a07097f9-2e2f-4b15-8170-774c6759a416">


same with using spread or replace as MergeDeepOption

Reply:
i think we have three options:

- find out why MergeDeep doesn't work (if you could provide playground link I can take a look)
- implement something similar to MergeDeep, like we did with UsableFields, that fits perfectly our use case (and is simpler than MergeDeep)
- no ts magic - just generate code that uses Omit

Reply:
> find out why MergeDeep doesn't work (if you could provide playground link I can take a look)

I created a [CodeSandbox](https://codesandbox.io/p/sandbox/hardcore-voice-k3w9cd?file=%2Fsrc%2Findex.ts%3A58%2C1-58%2C63). It seems the issue is related to the object being optional as the addressAsEmbeddable is working as desired.

> implement something similar to MergeDeep, like we did with UsableFields, that fits perfectly our use case (and is simpler than MergeDeep)

I still didn't start with https://github.com/type-challenges/type-challenges, so this will take me some time.

> no ts magic - just generate code that uses Omit

for me this is the best option, as this is the current state and the issue would be solved :-D

Reply:
> for me this is the best option, as this is the current state and the issue would be solved :-D

you know, this is the base for what future generator will generate and should be best possible practice....

Reply:
> you know, this is the base for what future generator will generate and should be best possible practice....

I think for now it's the best solution, but certainly can be improved. And should then also be updated in admin-generator.

I'm currently not experienced enough with type-level programming to either understand why MergeDeep type is not working the way I need it, or to create a custom type working with all situations currently required... (as the other pull-request already proved...)

Reply:
so, this is the last concern for this pull-request